packages/extension-chakra-storefront/src/hooks/use-auth-modal.js

```diff
@@ -275,6 +275,12 @@ export const AuthModal = ({
     const onBackToSignInClick = () =>
         initialView === PASSWORD_VIEW ? onClose() : setCurrentView(LOGIN_VIEW)
 
+    if (typeof window === 'undefined') {
+        // Render nothing on the server side. Otherwise, the Portal component would break SSR and its data fetching.
+        // Portal would call Ark UI's useEnvironmentContext hook, which tries to access `document` that doesn't exist on the server.
+        return
+    }
+
```

Comment:
Instead of having this one off logic, should we create a <ClientOnly> component..

Also there is this new <Island clientOnly> that might fit the bill to wrap the render portion of this render. Not that I really want to solidify the Island component, but it's there.

Reply:
Just trying to understand the behavior here - since the portal components are closed by default there shouldn't be any performance impact, correct?

Reply:
@unandyala Yeah I think there shouldn't be any performance impact. But what performance-specific scenario or metric were you thinking of?